src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -667,6 +667,10 @@ public function dumpFile(string $filename, $content)
             throw new \TypeError(sprintf('Argument 2 passed to "%s()" must be string or resource, array given.', __METHOD__));
         }
 
+        if (is_link($filename) && $linkTarget = $this->readlink($filename, true)) {
```

Comment:
This doesn't work if the symlink exists but is broken (but file_put_contents works with them)
Not sure how we should fix this. Any idea?

Reply:
In this case `readlink` will return `false` and `dumpFile` will thus overwrite the invalid symlink, will it not?

Reply:
Fixed in 0479d87791f24fc2222c45469d37f66c63bbe1b1